### PR TITLE
auth-backend: refactor OAuthHandlers to receive a single decorated request object

### PR DESCRIPTION
**Pull request by Rugvip**
Bit more forwards-compatible and slimmer interface. This also exposes the `state` object for `start` method to allow extensions, although it doesn't forward it for the `handler` yet.

### CODE REVIEW

File: plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts [hunk 1]
```diff
@@ -86,15 +86,12 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
     // set a nonce cookie before redirecting to oauth provider
     this.setNonceCookie(res, nonce);
 
-    const stateObject = { nonce: nonce, env: env };
-    const stateParameter = encodeState(stateObject);
+    const state = { nonce: nonce, env: env };
+    const forwardRed = Object.assign(req, { scope, state });
```
Comment (freben) [inline plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts:90]: req?

File: plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts [hunk 2]
```diff
@@ -185,8 +182,12 @@ export class OAuthAdapter implements AuthProviderRouteHandlers {
 
       const scope = req.query.scope?.toString() ?? '';
 
+      const forwardRed = Object.assign(req, { scope, refreshToken });
```
Comment (freben) [inline plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts:185]: same here?

**Review by freben — APPROVED** [review]
Not sure if gained super much by putting along with the request, but sure :)

**Comment by Rugvip** [comment]
Yeah not the most important part of this refactor. It is kinda nicely forwards-compatible in case we need to pass on the response object in the future, getting `(req, res)` instead of `(req, opts, res)`.

**Pull request merged**